### ISSUE TITLE
glfw3:x86-windows on VS17 : BUILD_FAILED

### ISSUE THREAD

**Issue by iondune** [Microsoft/vcpkg#852]
LOG
```
-- CURRENT_INSTALLED_DIR=S:/Programs/vcpkg/installed/x86-windows
-- DOWNLOADS=S:/Programs/vcpkg/downloads
-- CURRENT_PACKAGES_DIR=S:/Programs/vcpkg/packages/glfw3_x86-windows
-- CURRENT_BUILDTREES_DIR=S:/Programs/vcpkg/buildtrees/glfw3
-- CURRENT_PORT_DIR=S:/Programs/vcpkg/ports/glfw3/.
-- Using cached S:/Programs/vcpkg/downloads/glfw-3.2.1.zip
-- Testing integrity of cached file...
-- Testing integrity of cached file... OK
-- Extracting done
-- Configuring x86-windows-rel
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:14 (message):
    Command failed: S:/Programs/vcpkg/downloads/cmake-3.8.0-rc1-win32-x86/bin/cmake.exe;S:/Programs/vcpkg/buildtrees/glfw3/src/glfw-3.2.1;-DGLFW_BUILD_EXAMPLES=OFF;-DGLFW_BUILD_TESTS=OFF;-DGLFW_BUILD_DOCS=OFF;-DPACKAGE_CMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_LIST_DIR}/../..;-DBUILD_SHARED_LIBS=ON;-DCMAKE_CXX_FLAGS= /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP;-DCMAKE_C_FLAGS= /DWIN32 /D_WINDOWS /W3 /utf-8 /MP;-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON;-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON;-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON;-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE;-DCMAKE_VERBOSE_MAKEFILE=ON;-DVCPKG_APPLOCAL_DEPS=OFF;-DCMAKE_TOOLCHAIN_FILE=S:/Programs/vcpkg/scripts/buildsystems/vcpkg.cmake;-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON;-DCMAKE_CXX_FLAGS_RELEASE=/MD /O2 /Oi /Gy /DNDEBUG /Zi;-DCMAKE_C_FLAGS_RELEASE=/MD /O2 /Oi /Gy /DNDEBUG /Zi;-DCMAKE_SHARED_LINKER_FLAGS_RELEASE=/DEBUG /INCREMENTAL:NO /OPT:REF /OPT:ICF;-DCMAKE_EXE_LINKER_FLAGS_RELEASE=/DEBUG /INCREMENTAL:NO /OPT:REF /OPT:ICF;-G;Visual Studio 15 2017;-DCMAKE_BUILD_TYPE=Release;-DCMAKE_PREFIX_PATH=S:/Programs/vcpkg/installed/x86-windows;-DCMAKE_INSTALL_PREFIX=S:/Programs/vcpkg/packages/glfw3_x86-windows
    Working Directory: S:/Programs/vcpkg/buildtrees/glfw3/x86-windows-rel
    See logs for more information:
      S:\Programs\vcpkg\buildtrees\glfw3\config-x86-windows-rel-out.log
      S:\Programs\vcpkg\buildtrees\glfw3\config-x86-windows-rel-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:99 (vcpkg_execute_required_process)
  ports/glfw3/portfile.cmake:23 (vcpkg_configure_cmake)
  scripts/ports.cmake:73 (include)


Error: Building package glfw3:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: glfw3:x86-windows
  Vcpkg version: 0.0.77-784aa2dd619405596b4ae2277bc36a4a83c53535

Additionally, attach any relevant sections from the log files above.
```

config-x86-windows-rel-out.log
```
-- The C compiler identification is MSVC 19.10.25017.0
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.10.25017/bin/HostX86/x86/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.10.25017/bin/HostX86/x86/cl.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Looking for pthread.h
-- Looking for pthread.h - not found
-- Found Threads: TRUE  
-- Could NOT find Vulkan (missing:  VULKAN_LIBRARY VULKAN_INCLUDE_DIR) 
-- Using Win32 for window creation
-- Configuring incomplete, errors occurred!
See also "S:/Programs/vcpkg/buildtrees/glfw3/x86-windows-rel/CMakeFiles/CMakeOutput.log".
See also "S:/Programs/vcpkg/buildtrees/glfw3/x86-windows-rel/CMakeFiles/CMakeError.log".
```

config-x86-windows-rel-err.log
```
CMake Error at CMakeLists.txt:403 (install):
  install INSTALL(EXPORT) given keyword "EXPORT_LINK_INTERFACE_LIBRARIES",
  but target "glfw" does not have policy CMP0022 set to NEW.
```

CMakeError.log
```
Determining if the include file pthread.h exists failed with the following output:
Change Dir: S:/Programs/vcpkg/buildtrees/glfw3/x86-windows-rel/CMakeFiles/CMakeTmp

Run Build Command:"C:\Program Files (x86)\Microsoft Visual Studio\2017\Community/MSBuild/15.0/Bin/MSBuild.exe" "cmTC_37072.vcxproj" "/p:Configuration=Debug" "/p:VisualStudioVersion=15.0"
Microsoft (R) Build Engine version 15.1.548.43366

Copyright (C) Microsoft Corporation. All rights reserved.



Build started 3/24/2017 09:20:35.

Project "S:\Programs\vcpkg\buildtrees\glfw3\x86-windows-rel\CMakeFiles\CMakeTmp\cmTC_37072.vcxproj" on node 1 (default targets).

PrepareForBuild:

  Creating directory "cmTC_37072.dir\Debug\".

  Creating directory "S:\Programs\vcpkg\buildtrees\glfw3\x86-windows-rel\CMakeFiles\CMakeTmp\Debug\".

  Creating directory "cmTC_37072.dir\Debug\cmTC_37072.tlog\".

InitializeBuildStatus:

  Creating "cmTC_37072.dir\Debug\cmTC_37072.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.

VcpkgTripletSelection:

  Not using Vcpkg because VcpkgEnabled is "false"

ClCompile:

  C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.10.25017\bin\HostX86\x86\CL.exe /c /Zi /W3 /WX- /diagnostics:classic /MP /Od /Ob0 /Oy- /D WIN32 /D _WINDOWS /D "CMAKE_INTDIR=\"Debug\"" /D _MBCS /Gm- /RTC1 /MDd /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Fo"cmTC_37072.dir\Debug\\" /Fd"cmTC_37072.dir\Debug\vc141.pdb" /Gd /TC /analyze- /errorReport:queue   /utf-8 "S:\Programs\vcpkg\buildtrees\glfw3\x86-windows-rel\CMakeFiles\CMakeTmp\CheckIncludeFile.c"

  Microsoft (R) C/C++ Optimizing Compiler Version 19.10.25017 for x86

  Copyright (C) Microsoft Corporation.  All rights reserved.

  

  CheckIncludeFile.c

  cl /c /Zi /W3 /WX- /diagnostics:classic /MP /Od /Ob0 /Oy- /D WIN32 /D _WINDOWS /D "CMAKE_INTDIR=\"Debug\"" /D _MBCS /Gm- /RTC1 /MDd /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Fo"cmTC_37072.dir\Debug\\" /Fd"cmTC_37072.dir\Debug\vc141.pdb" /Gd /TC /analyze- /errorReport:queue   /utf-8 "S:\Programs\vcpkg\buildtrees\glfw3\x86-windows-rel\CMakeFiles\CMakeTmp\CheckIncludeFile.c"

  

S:\Programs\vcpkg\buildtrees\glfw3\x86-windows-rel\CMakeFiles\CMakeTmp\CheckIncludeFile.c(1): fatal error C1083: Cannot open include file: 'pthread.h': No such file or directory [S:\Programs\vcpkg\buildtrees\glfw3\x86-windows-rel\CMakeFiles\CMakeTmp\cmTC_37072.vcxproj]

Done Building Project "S:\Programs\vcpkg\buildtrees\glfw3\x86-windows-rel\CMakeFiles\CMakeTmp\cmTC_37072.vcxproj" (default targets) -- FAILED.



Build FAILED.



"S:\Programs\vcpkg\buildtrees\glfw3\x86-windows-rel\CMakeFiles\CMakeTmp\cmTC_37072.vcxproj" (default target) (1) ->

(ClCompile target) -> 

  S:\Programs\vcpkg\buildtrees\glfw3\x86-windows-rel\CMakeFiles\CMakeTmp\CheckIncludeFile.c(1): fatal error C1083: Cannot open include file: 'pthread.h': No such file or directory [S:\Programs\vcpkg\buildtrees\glfw3\x86-windows-rel\CMakeFiles\CMakeTmp\cmTC_37072.vcxproj]



    0 Warning(s)

    1 Error(s)



Time Elapsed 00:00:00.87
```

**Comment by ras0219-msft** [comment]
Thanks for reporting this!

I believe I've fixed this in e844167, which bumps glfw3 to version 3.2.1-1.

**Comment by iondune** [comment]
I can confirm it installed successfully, thank you!
